src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -15,125 +15,5311 @@
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_CHECK_PLAYER:
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        if (player->IsAlive() && player->IsInWorld() && player->hasQuest(QUEST_LOST_IN_THE_DARKNESS) && !player->IsQuestRewarded(QUEST_LOST_IN_THE_DARKNESS))
+                            _events.ScheduleEvent(EVENT_CHECK_PLAYER, 1s);
+                    }
+                    else
+                        me->DespawnOrUnsummon();
+
+                    break;
+                }
+
+                case EVENT_TALK_TO_PLAYER:
+                {
+                    Talk(TALK_0);
+                    break;
+                }
+
+                default:
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+};
+
+// Skitterweb Matriarch - 44906
+struct npc_silverpine_skitterweb_matriarch : public ScriptedAI
+{
+    npc_silverpine_skitterweb_matriarch(Creature* creature) : ScriptedAI(creature), _alreadyPulled(false) { }
+
+    void JustAppeared() override
+    {
+        _alreadyPulled = false;
+
+        me->SetDisableGravity(true);
+        me->SetHover(true);
+
+        me->SetAIAnimKitId(DATA_SPIDER_ANIMKIT1);
+
+        me->SetReactState(REACT_PASSIVE);
+
+        me->GetCreatureListWithEntryInGrid(_stalkerList, NPC_SKITTERWEB_STALKER, 8.0f);
+        for (Creature* stalker : _stalkerList)
+            me->CastSpell(stalker, SPELL_SKITTERWEB, true);
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        if (!_alreadyPulled)
+        {
+            _alreadyPulled = true;
+
+            me->SetAIAnimKitId(0);
+            me->PlayOneShotAnimKitId(DATA_SPIDER_ANIMKIT2);
+
+            _events.ScheduleEvent(EVENT_AGGRO_SPIDER, 2s + 500ms);
+        }
+        else
+        {
+            _events.ScheduleEvent(EVENT_SUMMON_SPIDERLINGS, 2s);
+            _events.ScheduleEvent(EVENT_VENOM_SPLASH, 4s, 7s);
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_AGGRO_SPIDER:
+                {
+                    me->SetDisableGravity(false);
+                    me->SetHover(false);
+
+                    me->GetMotionMaster()->MoveFall();
+
+                    _events.ScheduleEvent(EVENT_AGGRO_SPIDER + 1, 1s);
+                    break;
+                }
+
+                case EVENT_AGGRO_SPIDER + 1:
+                {
+                    for (Creature* stalker : _stalkerList)
```

Comment:
As far as I know, this may be unsafe. Are you sure that creature will not be removed from world after saving pointer to vector? It might be better to operate with ObjectGuid's.

Reply:
Stalkers do not get removed from world at any point, they're always there. In any case, I could change vector to guid ^^.